cmake/NVBenchDependencies.cmake

```diff
@@ -56,3 +56,36 @@ rapids_find_package(CUDAToolkit REQUIRED
 
 # Append CTK targets to this as we add optional deps (NMVL, CUPTI, ...)
 set(ctk_libraries CUDA::toolkit)
+
+################################################################################
+# CUDAToolkit -> NVML
+if (NVBench_ENABLE_NVML)
+  if (WIN32)
+    # The CUDA:: targets currently don't provide dll locations through the
+    # `IMPORTED_LOCATION` property, nor are they marked as `SHARED` libraries
+    # (they're currently `UNKNOWN`). This prevents the `nvbench_setup_dep_dlls`
+    # CMake function from copying the dlls to the build / install directories.
+    # This is discussed in https://gitlab.kitware.com/cmake/cmake/-/issues/22845
+    # and the other CMake issues it links to.
+    #
+    # We create a nvbench-specific target that configures the nvml interface as
+    # described here:
+    # https://gitlab.kitware.com/cmake/cmake/-/issues/22845#note_1077538
+    #
+    # Find the nvml.dll. It's somewhere under C:\Windows\System32.
+    find_library(NVBench_NVML_DLL nvml.dll REQUIRED
+      DOC "The full path to nvml.dll. Usually somewhere under C:/Windows/System32."
+    )
+    add_library(nvbench::nvml SHARED IMPORTED)
+    target_link_libraries(nvbench::nvml INTERFACE CUDA::toolkit)
+    set_target_properties(nvbench::nvml PROPERTIES
+      IMPORTED_LOCATION "${NVBench_NVML_DLL}"
+      IMPORTED_IMPLIB "${CUDA_nvml_LIBRARY}"
+    )
```

Comment:
@robertmaynard Can you check my WAR here and see if you spot anything amiss?

Reply:
I'm 100% sure this will break our export sets. I think I need to move this into a `FindNVML.cmake` file and use `rapids_find_package`. Does this sound right?

Also, any chance we can get `rapids_generate_find_module` to support DLLs? I know ya'll don't support dll platforms, so it's cool if not.

Reply:
I also hate how I'm finding `nvml.dll` -- it pretty much forces users to specify it manually. Unfortunately it's in a path with a non-obvious name, on my machine its:

`C:\Windows\System32\DriverStore\FileRepository\nvltui.inf_amd64_4d48d972b3eecbd5`

If you have any suggestions for how to find this automatically I'm definitely interested.

Reply:
> I think I need to move this into a `FindNVML.cmake` file

Or maybe it'd be best to modify `FindCUDAToolkit.cmake` and ship a custom version with nvbench.

Reply:
>  I think I need to move this into a FindNVML.cmake 

You are correct the logic needs to be replicated for consumers. CMake doesn't export IMPORTED targets, therefore for consumers of nvbench they will need to reconstruct the `nvbench::nvml` target.

You could move the logic to `NVBenchNVML.cmake`, install that file, and include it from the `nvbench-config.cmake` file. Since shipping a custom FindNVML.cmake might cause problems if NVML ever starts to
ship a `NVML-config.cmake`

Reply:
> I also hate how I'm finding nvml.dll -- it pretty much forces users to specify it manually. Unfortunately it's in a path with a non-obvious name....

Reading https://github.com/nvpro-samples/nvml_enterprise_gpu_check#loading-nvml it looks like if you also issue a `find_package(OpenGL)` you can use that location to find nvml.dll ( on windows only ).

Reply:
@robertmaynard Thanks for the tips, I got this working :D